### ISSUE TITLE
Deleting object properties doesn't update "ordered" data

### ISSUE THREAD

**Issue by Keltin42** [prescottprue/redux-firestore#243]
**What is the current behavior?**
When you delete a field from a firebase record, the change is not correctly propagated to the ordered data structure.  The change DOES show up in the firestore.data structure stored in redux, and changes to add or modify fields propagate to firestore.ordered, but deleted fields do not propagate.

**What is the expected behavior?**
The ordered data structure should also reflect the same changes.

**Which version of redux-firestore are you using? What about other dependencies?**
I was on these versions and saw the issue:
    "firebase": "6.6.0",
    "react": "^16.8.6",
    "react-dom": "^16.8.6",
    "react-filepond": "^7.0.1",
    "react-modal": "^3.10.1",
    "react-redux": "5.1.1",
    "react-redux-firebase": "2.4.1",
    "react-scripts": "3.1.1",
    "recompose": "^0.30.0",
    "redux": "^4.0.1",
    "redux-firestore": "0.9.0"

I just upgraded my project hoping there would be a fix for this issue, but the issue is still there in the updated versions:
    "firebase": "7.2.0",
    "react": "^16.8.6",
    "react-dom": "^16.8.6",
    "react-filepond": "^7.0.1",
    "react-modal": "^3.10.1",
    "react-redux": "6.0.0",
    "react-redux-firebase": "3.0.3",
    "react-scripts": "3.1.1",
    "recompose": "^0.30.0",
    "redux": "^4.0.1",
    "redux-firestore": "0.9.0"

**History of this issue**

I have seen previous bugs around this issue:  https://github.com/prescottprue/redux-firestore/issues/71

I believe the issue may be right around here:
https://github.com/prescottprue/redux-firestore/blob/321578336bf907e06cbbe4c9c271321b04bd4aee/src/reducers/dataReducer.js#L62

Your previous fix (https://github.com/prescottprue/redux-firestore/pull/72/files#diff-8a73042446f050ebb83f87e2ea9cc125R52-R53) added an assign, but that code seems to have disappeared.  The comment about setWith() is correct, but the merge will not handle deleted fields correctly.

**Minimal demo to reproduce issue (using [codesandbox](https://codesandbox.io/) or similar)**

Code to initiate the change:
```
            props.firestore
                .runTransaction(t => {
                    return t.get(itemRef).then(doc => {
                        const newItemInfo = {
                            imageUrl: props.firestore.FieldValue.delete(),
                            foo: 'bar2'
                        };

                        t.update(itemRef, newItemInfo);
                        t.update(historyItemRef, newItemInfo);
                    });
                })
```

The redux debugger update:
![image](https://user-images.githubusercontent.com/33882419/67789905-06266880-fa32-11e9-82e8-b2082e0312cb.png)

**Comment by prescottprue** [comment]
[This PR for v0.10.0](https://github.com/prescottprue/redux-firestore/pull/244) switches back to not merging, so it fixes this issue. In order to handle the case of subcollections, I added a `console.error` for queries that have `subcollections` that are not using `storeAs` - this seems like the best way to accomplish both for now as well as prep for v1 which stores all collections (even subcollections) at the top level of state

**Comment by Keltin42** [comment]
I don't understand your comment about not using `storeAs`.  I am definitely using it, here is my `firestoreConnect` result:

            {
                collection: 'shoppinglists',
                doc: props.shoppinglistId,
                subcollections: [{ collection: 'items' }],
                storeAs: 'currentItemList'
            }

You can also see the `storeAs` in redux path in the screenshot.

**Comment by prescottprue** [comment]
You are supposed to be using `storeAs` - what I was getting at is that I added a console.error for folks not using`storeAs` with subcollections as this will cause issues with the change I made to fix the issue you mentioned (with fields not being removed on a document).

**Comment by Keltin42** [comment]
Ahhh.  I get it now.  Thanks!  :)

**Comment by prescottprue** [comment]
Fixed in the [`v0.10.0` release](https://github.com/prescottprue/redux-firestore/releases/tag/v0.10.0) and a test was added for updating of array items in state.  Reach out if things aren't working as expected, thanks for reporting!
